src/lib/tests/symlink-manage.test.ts

```diff
@@ -0,0 +1,195 @@
+			expect( fs.watch ).toHaveBeenCalledWith( mockProjectPath, expect.any( Object ) );
+
+			await symlinkManager.stopWatching();
+			await watchPromise;
+
+			expect( ( symlinkManager as unknown as SymlinkManagerPrivateProperties ).watcher ).toBeNull();
+			expect(
+				( symlinkManager as unknown as SymlinkManagerPrivateProperties ).watcherAbortController
+			).toBeNull();
+		} );
+	} );
+
+	describe( 'file change handling', () => {
+		it( 'should handle symlink creation', async () => {
+			mockPHP.fileExists.mockImplementation( ( path ) => ! path.includes( 'vfspath' ) );
+			mockPHP.readlink.mockImplementation( ( path ) => '/mock/document/root/vfspath/' + path );
+			( pathExists as jest.Mock ).mockResolvedValue( true );
+			const mockWatcher = {
+				[ Symbol.asyncIterator ]: jest.fn().mockImplementation( () => ( {
+					next: jest
+						.fn()
+						.mockResolvedValueOnce( {
+							done: false,
+							value: { eventType: 'rename', filename: 'newSymlink' },
+						} )
+						.mockResolvedValueOnce( { done: true, value: undefined } ),
+				} ) ),
+			};
+			jest.spyOn( fs, 'watch' ).mockReturnValue( mockWatcher as never );
+
+			( fs.lstat as jest.Mock ).mockResolvedValue( { isSymbolicLink: () => true } as Stats );
+			( fs.realpath as jest.Mock ).mockResolvedValue( '/real/path/to/newSymlink' );
+
+			const watchPromise = symlinkManager.startWatching();
+
+			await new Promise( ( resolve ) => setTimeout( resolve, 100 ) );
+
+			await symlinkManager.stopWatching();
+			await watchPromise;
+
+			expect( mockPHP.mkdir ).toHaveBeenCalled();
+			expect( mockPHP.mount ).toHaveBeenCalled();
+			const privateProps = symlinkManager as unknown as SymlinkManagerPrivateProperties;
+			expect( privateProps.symlinks.has( 'newSymlink' ) ).toBe( true );
+		} );
+
+		it( 'should handle symlink deletion', async () => {
+			const privateProps = symlinkManager as unknown as SymlinkManagerPrivateProperties;
+			privateProps.symlinks.set( 'existingSymlink', '/vfs/path/to/target' );
+			const unmountfn = jest.fn();
+			privateProps.mountedTargets.set( '/vfs/path/to/target', {
+				referenceCount: 1,
+				unmountFunction: unmountfn,
+			} );
+
+			const mockWatcher = {
+				[ Symbol.asyncIterator ]: jest.fn().mockImplementation( () => ( {
+					next: jest
+						.fn()
+						.mockResolvedValueOnce( {
+							done: false,
+							value: { eventType: 'rename', filename: 'existingSymlink' },
+						} )
+						.mockResolvedValueOnce( { done: true, value: undefined } ),
+				} ) ),
+			};
+			jest.spyOn( fs, 'watch' ).mockReturnValue( mockWatcher as never );
+
+			( pathExists as jest.Mock ).mockResolvedValue( false );
+
+			const watchPromise = symlinkManager.startWatching();
+
+			await new Promise( ( resolve ) => setTimeout( resolve, 100 ) );
+
+			await symlinkManager.stopWatching();
+			await watchPromise;
+
+			expect( privateProps.symlinks.has( 'existingSymlink' ) ).toBe( false );
+			expect( unmountfn ).toHaveBeenCalled();
```

Comment:
Similar to my comment above:

```suggestion
			await symlinkManager.startWatching();
			await waitFor( () => {
				expect( privateProps.symlinks.has( 'existingSymlink' ) ).toBe( false );
				expect( unmountfn ).toHaveBeenCalled();
			} );
			await symlinkManager.stopWatching();
```